src/Microsoft.IdentityModel.Tokens/SymmetricSignatureProvider.cs

```diff
@@ -242,15 +246,74 @@ public override bool Verify(byte[] input, byte[] signature)
         /// <exception cref="ObjectDisposedException"><see cref="Dispose(bool)"/> has been called.</exception>
         /// <exception cref="InvalidOperationException">If the internal <see cref="KeyedHashAlgorithm"/> is null. This can occur if a derived type deletes it or does not create it.</exception>
         public bool Verify(byte[] input, byte[] signature, int length)
+        {
+            if (signature == null || signature.Length == 0)
+                throw LogHelper.LogArgumentNullException(nameof(signature));
+
+            if (input == null || input.Length == 0)
+                throw LogHelper.LogArgumentNullException(nameof(input));
+
```

Comment:
nit: if the length check has been moved to the other Verify(), these two can be skipped too as all of them are being checked.

Reply:
Fixed